src/sentry/static/sentry/app/utils/performance/quickTrace/quickTraceQuery.tsx

```diff
@@ -1,131 +1,61 @@
 import React from 'react';
-import omit from 'lodash/omit';
 
-import {Client} from 'app/api';
-import {getTraceDateTimeRange} from 'app/components/events/interfaces/spans/utils';
-import {ALL_ACCESS_PROJECTS} from 'app/constants/globalSelectionHeader';
-import {Event, EventTransaction} from 'app/types/event';
-import EventView from 'app/utils/discover/eventView';
-import GenericDiscoverQuery, {
-  DiscoverQueryProps,
-  GenericChildrenProps,
-} from 'app/utils/discover/genericDiscoverQuery';
-import withApi from 'app/utils/withApi';
+import {DiscoverQueryProps} from 'app/utils/discover/genericDiscoverQuery';
+import TraceFullQuery from 'app/utils/performance/quickTrace/traceFullQuery';
+import TraceLiteQuery from 'app/utils/performance/quickTrace/traceLiteQuery';
+import {
+  TraceLiteQueryChildrenProps,
+  TraceProps,
+} from 'app/utils/performance/quickTrace/types';
+import {flattenRelevantPaths} from 'app/utils/performance/quickTrace/utils';
 
-import {isTransaction} from './utils';
+type RequestProps = DiscoverQueryProps & TraceProps;
 
-export type EventLite = {
-  event_id: string;
-  span_id: string;
-  transaction: string;
-  'transaction.duration': number;
-  project_id: number;
-  project_slug: string;
-  parent_event_id: string | null;
-  parent_span_id: string | null;
-  is_root: boolean;
+export type QuickTraceQueryChildrenProps = TraceLiteQueryChildrenProps & {
```

Comment:
I'm wondering if you can make use of discriminated unions for this type. Described here https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions

```
type FullTrace = {
  type: 'full';
} & TraceFullQueryChildrenProps;

type PartialTrace = {
  type: 'partial';
  trace: TraceLite;
};

type EmptyTrace = {
  type: 'empty';
  trace: TraceLite;
};

type TraceResults = FullTrace | PartialTrace | EmptyTrace;

type QuickTraceQueryChildrenProps = Omit<
  GenericChildrenProps<TraceProps>,
  'tableData' | 'pageLinks'
> & TraceResults;
```